### PR TITLE
Search

### PR DESCRIPTION
**Pull request by rawalparas**
I created a search model to insert books data into it.

### CODE REVIEW

File: src/model/authorModel.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const regex = require('../helper/regexPatterns.js');
 const validateSchemaName = require('../helper/helperMethods.js')
 
 const authorSchema = new mongoose.Schema({
-    name : {
+    name: {
```
Comment (athanbhanit) [inline src/model/authorModel.js:6]: make it unique

File: src/API/controller/create/createController.js [hunk 2]
```diff
@@ -22,9 +23,37 @@ module.exports = {
...
     } catch (err) {
       return res.status(400).send(err.message)
     }
-  },
-};
\ No newline at end of file
+  }
+};
+
+async function findData(Object_id){
```
Comment (athanbhanit) [inline src/API/controller/create/createController.js:48]: What happen if any error occured in given method.

File: src/API/controller/create/createController.js [hunk 3]
```diff
@@ -22,9 +23,37 @@ module.exports = {
...
+  return await search.findOne({
```
Comment (athanbhanit) [inline src/API/controller/create/createController.js:49]: What happen if any error occured in given method.

File: src/model/bookModel.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ const bookSchema = new mongoose.Schema({
     name: {
       type: String,
       required: true,
+      unique : [true , "Duplicate Book Found"],
```
Comment (athanbhanit) [inline src/model/bookModel.js:9]: install mentioned lib for getting custom message, and check if it works.  npm install mongoose-unique-validator

File: src/API/controller/create/createController.js [hunk 5]
```diff
@@ -1,30 +1,52 @@
...
-        authorId = await author.create({ name: authorName });
-      }
-      bookData = await book.create({
-          name: bookName,
-          authorId: authorId._id,
-          languageId: languageId._id,
-        });
       return res.status(200).send(messages.SUCCESSSFULLY_CREATED);
     } catch (err) {
       return res.status(400).send(err.message)
```
Comment (athanbhanit) [inline src/API/controller/create/createController.js:31]: Why 400?

File: src/helper/messages.js [hunk 6]
```diff
@@ -18,6 +18,7 @@ const INTERNAL_SERVER_ERROR = "Internal Server Error";
 const SUCCESSSFUL_GET = "Successfully getting the data";
 const EMPTY_COLLECTION = "Collection is empty"
 const REGEX_PATTERN_UNMATCHED = "Field does not match the required pattern";
+const MUST_BE_UNIQUE = "Value must be unique";
```
Comment (athanbhanit) [inline src/helper/messages.js:21]: Book must be unique

File: package-lock.json [hunk 7]
```diff
@@ -439,6 +440,21 @@
         "node": ">=12.0.0"
       }
     },
+    "node_modules/lodash.foreach": {
```
Comment (athanbhanit) [inline package-lock.json:443]: What are these changes.

File: src/helper/messages.js (hunk 6 above)
Comment (rawalparas) [reply src/helper/messages.js:21]: It is solved.

File: src/helper/messages.js [hunk 8]
```diff
@@ -18,6 +18,7 @@ const INTERNAL_SERVER_ERROR = "Internal Server Error";
 const SUCCESSSFUL_GET = "Successfully getting the data";
 const EMPTY_COLLECTION = "Collection is empty"
 const REGEX_PATTERN_UNMATCHED = "Field does not match the required pattern";
+const BOOKNAME_MUST_BE_UNIQUE = "BookName must be unique";
```
Comment (athanbhanit) [inline src/helper/messages.js:21]: Book Name should have space and Name starts with small case

**Review by athanbhanit — APPROVED** [review]

**Pull request merged**